mobile/salto/src-tauri/src/background_tasks.rs

```diff
@@ -0,0 +1,63 @@
+use std::{sync::Arc, time::Duration};
+use tauri::Emitter;
+use tokio::sync::RwLock;
+
+use crate::{PriceConfig, PriceResponce};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
+    #[error(transparent)]
+    Tauri(#[from] tauri::Error),
+}
+
+async fn fetch_and_emit_prices(url: &str, app: &tauri::AppHandle) -> Result<(), Error> {
+    let resp = reqwest::get(url).await?;
+    let new_prices = resp.json::<PriceResponce>().await?;
+    app.emit("new-price", new_prices)?;
+    Ok(())
+}
+
+pub async fn start_price_fetcher(config: Arc<RwLock<PriceConfig>>, app_thread: tauri::AppHandle) {
+    let mut interval = tokio::time::interval(Duration::from_secs(10));
+
+    loop {
+        let cfg = config.read().await;
+
+        if !cfg.assets.is_empty() && !cfg.currencies.is_empty() {
+            break;
+        }
```

Comment:
yeah. Don't wait for the fronted to call the back to get the config.
Just get it yourself there based on the database content